src/main/java/edu/unc/lib/boxc/migration/cdm/services/FindingAidReportService.java

```diff
@@ -102,6 +117,146 @@ public Path getHookIdReportPath() {
+        CdmFieldInfo fieldInfo = fieldService.loadFieldsFromProject(project);
+        List<String> exportFields = fieldInfo.listAllExportFields();
+
+        for (String field : fieldsToReport) {
+            int fieldRecords = 0;
+            if (exportFields.contains(field)) {
+                String query = "select count(*) from " + CdmIndexService.TB_NAME + " where " + field + " != ''";
+                fieldRecords = countRecords(query);
+            }
+            outputLogger.info(DisplayProgressUtil.displayProgressWithLabel(field, fieldRecords, totalRecords));
+        }
+    }
+
+    /**
+     * List of unique collection id values
+     */
+    private List<String> listCollectionIds() {
+        String collectionIdQuery = "select distinct " + FindingAidService.DESCRI_FIELD
+                + " from " + CdmIndexService.TB_NAME;
+        List<String> uniqueCollectionIds = new ArrayList<>();
+
+        try (Connection conn = indexService.openDbConnection()) {
+            var stmt = conn.createStatement();
+            var rs = stmt.executeQuery(collectionIdQuery);
+            while (rs.next()) {
+                if (!rs.getString(1).isBlank()) {
+                    uniqueCollectionIds.add(rs.getString(1));
+                }
+            }
+            return uniqueCollectionIds;
+        } catch (SQLException e) {
+            throw new MigrationException("Error interacting with export index", e);
+        }
+    }
+
+    /**
+     * Count records for a given query
+     */
+    private Integer countRecords(String query) {
+        try (Connection conn = indexService.openDbConnection()) {
+            var stmt = conn.createStatement();
+            var rs = stmt.executeQuery(query);
+            rs.next();
+            return rs.getInt(1);
+        } catch (SQLException e) {
+            throw new MigrationException("Error interacting with export index", e);
+        }
+    }
+
+    /**
+     * Map of potential collection id values and the fields they come from
+     */
+    private Map<String, List<String>> listPotentialCollectionIds() {
+        fieldService.validateFieldsFile(project);
+        CdmFieldInfo fieldInfo = fieldService.loadFieldsFromProject(project);
+        List<String> exportFields = fieldInfo.listAllExportFields();
+
+        Map<String, List<String>> potentialCollectionIds = new HashMap<>();
+        // only collect the first 10 potential ids
+        try (Connection conn = indexService.openDbConnection()) {
+            var stmt = conn.createStatement();
+            for (String field : exportFields) {
+                List<String> ids = new ArrayList<>();
+                ResultSet rs = stmt.executeQuery(" select " + field + " from " + CdmIndexService.TB_NAME
+                        + " where " + field + " like " + "'^[A-Za-z0-9]{5}-?z?$' limit 10");
+                while (rs.next()) {
+                    ids.add(rs.getString(1));
+                }
+                potentialCollectionIds.put(field, ids);
```

Comment:
I think there might need to be a check to see if ids is empty, otherwise the map will never be empty